drafts/str4d-overwinter-activation/draft1.rst

```diff
@@ -0,0 +1,237 @@
+height. This can handled in the same way as any regular chain orphaning or reorganization, as long as the new
+chain is valid over the same epochs.
+
+Post-activation upgrading
+`````````````````````````
+If a user does not upgrade their node to a compatible software version before ``ACTIVATION_HEIGHT`` is
+reached, their node will follow any pre-upgrade branch that persists, and may download blocks that are
+incompatible with the post-upgrade branch. If the user subsequently upgrades their node to a compatible
+software version, the node will consider these blocks to be invalid, and MUST take one of the two following
+actions:
+
+- Discard all blocks of height ``ACTIVATION_HEIGHT`` and above, and then synchronize with the network.
+
+- Shut down and alert the user of the issue. In this case, the node could offer an option to perform the first
+  action.
+
+Memory pool
+-----------
+
+While the current chain tip height is below ``ACTIVATION_HEIGHT``, nodes SHOULD NOT accept transactions that
+will only be valid on the post-upgrade branch.
+
+When the current chain tip height reaches ``ACTIVATION_HEIGHT``, the node's local transaction memory pool
+SHOULD be cleared of transactions that will never be valid on the post-upgrade branch.
+
+Two-way replay protection
+-------------------------
+
+Before the Overwinter network upgrade, two-way replay protection is ensured by enforcing post-upgrade that the
+MSB of the transaction version is set to 1. [#zip-tx-format]_ From the perspective of old nodes, the
+transactions will have a negative version number, which is invalid under the old consensus rules. Enforcing
+this rule trivially makes old transactions invalid on the Overwinter branch.
+
+After the Overwinter network upgrade, two-way replay protection is ensured by transaction signatures
+committing to a specific ``BRANCH_ID``. [#zip-0143]_
+
+Wipe-out protection
+-------------------
+
+Nodes running upgrade-aware software versions will enforce the upgraded consensus rules from
+``ACTIVATION_HEIGHT``. The chain from that height will not reorg to a pre-upgrade branch if any block in that
+branch would violate the new consensus rules (such as including any old-format transaction).
+
+Care must be taken, however, to account for possible edge cases where the old and new consensus rules do not
+differ. For example, if the non-upgraded chain only contained empty blocks from ``ACTIVATION_HEIGHT``, and the
+coinbase transactions were valid under both the old and new consensus rules, a wipe-out could occur. The
+Overwinter network upgrade is not susceptible to this because all previous transaction versions will become
+invalid, meaning that the coinbase transactions must use the newer transaction version. More generally, this
+issue could be addressed in a future network upgrade by modifying the block header to include a commitment to
+the ``BRANCH_ID``.
+
+
+Example
+=======
+
+TBC
```

Comment:
Do we need examples here? In e.g. #133 we'd want test vectors with serialized transactions to parse, but here I'm not sure there's anything useful that isn't implementation-specific (e.g. the C++ unit tests depend on the C++ implementation's chosen API).

Reply:
I don't think we do need this section.